File: library/src/main/java/com/getstream/sdk/chat/rest/request/SendActionRequest.java
package com.getstream.sdk.chat.rest.request;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.Map;

public class SendActionRequest {
    @SerializedName("id")
    @Expose
    String channelId;

    @SerializedName("message_id")
    @Expose
    String messageId;

    @SerializedName("type")
    @Expose
    String type;

    @SerializedName("form_data")
    @Expose
    Map<String, String> formData;

    public SendActionRequest(String channelId, String messageId, String type, Map form_data) {
        this.channelId = channelId;
        this.messageId = messageId;
        this.type = type;
        this.formData = form_data;
    }
}

File: library/src/main/java/io/getstream/chat/android/client/api/models/SendActionRequest.kt
package io.getstream.chat.android.client.api.models

import com.google.gson.annotations.SerializedName


data class SendActionRequest(
    val channelId: String,
    val messageId: String,
    val type: String,
    @SerializedName("form_data")
    val formData: Map<Any, Any>
)